### PR TITLE
bump to golang 1.20

### PR DESCRIPTION
**Pull request by arkodg**

### CODE REVIEW

**Comment by arkodg** [comment]
okay this is odd, I typed the `go-version` to `1.20`, but `setup-go` thinks the version is `1.2`

**Comment by zirain** [comment]
seems a bug of regex?

**Comment by arkodg** [comment]
> seems a bug of regex?

looks like single quotes`1.20` helped

**Comment by arkodg** [comment]
same issue as https://github.com/envoyproxy/gateway/pull/1063#issuecomment-1436428192

File: tools/src/golangci-lint/go.mod [hunk 1]
```diff
@@ -1,12 +1,13 @@
 module local
 
-go 1.19
+go 1.20
 
-require github.com/golangci/golangci-lint v1.50.1
+require github.com/golangci/golangci-lint v1.51.2
```
Comment (zirain) [inline tools/src/golangci-lint/go.mod:5]: do not upgrade `golangci` for now？

Comment (arkodg) [reply tools/src/golangci-lint/go.mod:5]: it gets killed, check out this run from a previous commit in this PR
https://github.com/envoyproxy/gateway/actions/runs/4368188859/jobs/7640450648#step:5:114

Comment (zirain) [reply tools/src/golangci-lint/go.mod:5]: my local machine got OOMed, after upgrade to go1.20.2 with no changes in golangci-lint.

**Comment by zirain** [comment]
seems related to https://github.com/dominikh/go-tools/commit/d717045480130461e1e777ead7f2981621f0d9f0

**Comment by zirain** [comment]
just need wait bot update golangci-lint to `1.51.3`

File: .golangci.yml [hunk 2]
```diff
@@ -38,6 +38,11 @@ linters-settings:
     simplify: true
   unparam:
     check-exported: false
+  revive:
+    rules:
+      # TODO: enable if-return check
```
Comment (zirain) [inline .golangci.yml:43]: xref: https://github.com/envoyproxy/gateway/issues/1185

**Comment by arkodg** [comment]
@Xunzhuo @AliceProxy can you ptal

**Review by Xunzhuo — APPROVED** [review]
🚀

**Review by AliceProxy — APPROVED** [review]
:+1:

**Pull request merged**